devito/passes/clusters/asynchrony.py

```diff
@@ -237,10 +237,11 @@ def _actions_from_update_memcpy(c, d, clusters, actions, sregistry):
     target = e.lhs.function
 
     fetch = e.rhs.indices[d]
-    if direction is Forward:
-        findex = fetch + 1
+    fshift = 1 if direction is Forward else -1
```

Comment:
random thought: we should just set the direction values to -1, 0 (Any), 1

Reply:
Yeah good point. 0 might lead to scheduling issues though